watertap/costing/unit_models/clarifier.py

```diff
@@ -0,0 +1,361 @@
+        cost_primary_clarifier(blk, **kwargs)
+    else:
+        raise ConfigurationError(
+            f"{blk.unit_model.name} received invalid argument for clarifier_type:"
+            f" {clarifier_type}. Argument must be a member of the ClarifierType Enum."
+        )
+
+
+def build_circular_clarifier_cost_param_block(blk):
+
+    blk.construction_a_parameter = pyo.Param(
+        initialize=-6e-4,
+        doc="A parameter for construction cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**4,
+    )
+
+    blk.construction_b_parameter = pyo.Param(
+        initialize=98.952,
+        doc="B parameter for construction cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**2,
+    )
+
+    blk.construction_c_parameter = pyo.Param(
+        initialize=191806,
+        doc="C parameter for construction cost",
+        units=pyo.units.USD_2011,
+    )
+
+    blk.O_and_M_a_parameter = pyo.Param(
+        initialize=8e-10,
+        doc="A parameter for operation and maintenance cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**6 / pyo.units.year,
+    )
+
+    blk.O_and_M_b_parameter = pyo.Param(
+        initialize=-5e-5,
+        doc="B parameter for operation and maintenance cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**4 / pyo.units.year,
+    )
+
+    blk.O_and_M_c_parameter = pyo.Param(
+        initialize=1.6945,
+        doc="C parameter for operation and maintenance cost",
+        units=pyo.units.USD_2011 / pyo.units.ft**2 / pyo.units.year,
+    )
+
+    blk.O_and_M_d_parameter = pyo.Param(
+        initialize=7207,
+        doc="D parameter for operation and maintenance cost",
+        units=pyo.units.USD_2011 / pyo.units.year,
+    )
+
+    blk.electricity_percentage = pyo.Param(
+        initialize=0.04,
+        doc="Percentage of electricity cost over O & M cost",
+        units=pyo.units.dimensionless,
+    )
+
+
+@register_costing_parameter_block(
+    build_rule=build_circular_clarifier_cost_param_block,
+    parameter_block_name="circular",
+)
+def cost_circular_clarifier(blk, cost_electricity_flow=True):
+    """
+    Circular clarifier costing method [1]
+    """
+    make_capital_cost_var(blk)
+    make_fixed_operating_cost_var(blk)
+
+    surface_area = pyo.units.convert(
+        blk.unit_model.surface_area, to_units=pyo.units.ft**2
+    )
+
+    blk.capital_cost_constraint = pyo.Constraint(
+        expr=blk.capital_cost
+        == pyo.units.convert(
+            blk.costing_package.circular.construction_a_parameter * surface_area**2
+            + blk.costing_package.circular.construction_b_parameter * surface_area
+            + blk.costing_package.circular.construction_c_parameter,
+            to_units=blk.costing_package.base_currency,
+        )
+    )
+
+    max_surface_area_limit = 200 * pyo.units.ft**2
+    if pyo.value(surface_area) > pyo.value(max_surface_area_limit):
+        num_O_and_M = math.ceil(
+            pyo.value(surface_area) / pyo.value(max_surface_area_limit)
+        )
```

Comment:
Oh, wait a second- this can be problematic. Surface area is a variable, so when unfixed, this value for surface area will be changing with solver iterations.

Reply:
So, how can I revise this? I think the surface area should be fixed for solving the costing problem, and once it's value is larger than the limit, we have to change to another option for costing this unit

Reply:
Actually, the underlying issue here is that once you go beyond the valid range for surface area, as stated in the literature, the assumption is that you would need to add on another clarifier. Thus, there can be _n_ clarifiers, and one might think that you'd need integer programming to deal with this. 

I am facing a similar issue with the thickener now.

However, I think we have some options. First, if you wanted to have a conditional like this, it'd have to be something like and `Expr_If `or reworked as a `smooth_max `or `smooth_min ` approach, since surface area can't also be guaranteed to be fixed, and more than that, I would argue that surface area shouldn't be fixed. Rather it should be a decision variable.

At this point, I might lean towards holding off on these O&M costs (or leave them with the intention of changing in next PR) since we want to be careful about how we implement usage of _n_ number of units.

The other thing to consider is that we can calculate the effective volume (i.e., the total volume of all clarifier units) based on (1) HRT* Qin and (2) surface_area * effective height, where effective height would account for _n_ number of units implicitly. (Imagine stacking all of our clarifiers on top of each other, each with the same diameter; this would give us the total volume of all reactors and sidestep using _n_.) This is the approach I am considering now. 

As your capital cost calculation stands now, it isn't clear that we'd be computing the total cost for all clarifiers needed, if we limit surface area to that of what one clarifier would typically have.

Reply:
On the other hand, there would be no penalty for the optimizer in terms of allowing effective height to go as high as possible to minimize surface area and thus clarifier capital cost. Perhaps this is where O&M costs can come in and serve as that penalty. That is, minimizing surface are would lead to large effective height and thus more individual units, and if O&M costs are driven up by the number of individual units, then that impose some sort of limit on minimizing surface area. Still thinking this through. . .

Reply:
Yeah, it's a good point. I think I can take off this fixe O&M cost first for this PR. For circular clarifier, the upper limit is very low (200 ft^2) and only account for O&M cost, but the construction cost relationship is valid for (0 to 32000 ft^2). So I guess the limit for O&M cost does not mean we have multiple clarifiers for this case, just the O&M cost relationship is valid for membrane exchange within 200 ft^2 and we have to do multiple operations.